src/tests/test_faida.cpp

```diff
@@ -0,0 +1,227 @@
+#include <filesystem>
+
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/ind/faida/faida.h"
+#include "config/names.h"
+#include "datasets.h"
+
+namespace fs = std::filesystem;
+
+namespace tests {
+
+using IndCCTest = std::pair<int, std::vector<int>>;
+using IndsTest = std::set<std::pair<IndCCTest, IndCCTest>>;
+using DatasetsOrderTest = std::vector<std::string>;
+
+void CheckINDsListsEquality(algos::INDAlgorithm::INDList const& actual, IndsTest const& expected) {
+    ASSERT_EQ(actual.size(), expected.size())
+            << "count of generated dependencies does not match: expected " << expected.size()
+            << ", got " << actual.size();
+
+    auto to_test_cc = [](model::IND::ColumnCombination const& cc) -> IndCCTest {
+        return std::make_pair(cc.GetTableIndex(), cc.GetColumnIndices());
+    };
+
+    for (auto const& dep : actual) {
+        if (expected.find(std::make_pair(to_test_cc(dep.GetLhs()), to_test_cc(dep.GetRhs()))) ==
+            expected.end()) {
+            FAIL() << "generated dep '" << dep.ToString() << "' is not present in expected";
+        }
+    }
+    SUCCEED();
+}
+
+void CheckResultContainsINDs(algos::INDAlgorithm::INDList const& actual, IndsTest expected_subset) {
+    ASSERT_NE(actual.size(), expected_subset.size())
+            << "count of generated dependencies must not not be equal to the subset size: got "
+            << actual.size();
+
+    auto to_test_cc = [](model::IND::ColumnCombination const& cc) -> IndCCTest {
+        return std::make_pair(cc.GetTableIndex(), cc.GetColumnIndices());
+    };
+
+    for (auto const& dep : actual) {
+        auto iter = expected_subset.find(
+                std::make_pair(to_test_cc(dep.GetLhs()), to_test_cc(dep.GetRhs())));
+
+        if (iter != expected_subset.end()) {
+            expected_subset.erase(iter);
+        }
+    }
+    ASSERT_EQ(expected_subset.size(), 0);
+}
+
+void CheckTablesInfoEquality(algos::INDAlgorithm::DatasetsOrder const& actual,
+                             DatasetsOrderTest const& expected) {
+    ASSERT_EQ(actual.size(), expected.size()) << "count of datasets does not match: expected "
+                                              << expected.size() << ", got " << actual.size();
+
+    for (unsigned table_id = 0; table_id != expected.size(); ++table_id) {
+        if (expected[table_id] != actual[table_id].table_name) {
+            FAIL() << "Expected table name '" << expected[table_id] << "', but received '"
+                   << actual[table_id].table_name << "'";
+        }
+    }
+    SUCCEED();
+}
+
+class INDAlgorithmTest : public ::testing::Test {
```

Comment:
It might make sense to note the motivation for creating a separate file (about the approximate algorithm) in this file, for example in the description of a class for a test. This will help people who will learn code in the future.